### PR TITLE
Update .NET Core to 3.1, and update NuGet Packages

### PR DESCRIPTION
**Pull request by ameisen**
This updates the .NET Core framework used from 3.0 to 3.1, and also updates all NuGet packages to their latest non-preview versions.

.NET Core 3.0 support ended March 3, 2020.

### CODE REVIEW

**Comment by jduncanator** [comment]
Thanks! .NET Core 3.1 update is long overdue.

Is there a reason for updating the NuGet packages? I’m usually of the belief that, unless we need a feature or something’s broken, updating NuGet packages adds a tonne of additional testing work that needs to be done.

**Comment by ameisen** [comment]
Not particularly; I am of the habit of keeping the packages up to date. I've been running them up to date for a while now with no particular issues. I can remove the NuGet updates if you'd like.

In regards to `System.Runtime.CompilerServices.Unsafe`, I do believe that it's generally recommended to keep that up to date with the framework, since it's a runtime extension.

**Comment by jduncanator** [comment]
I don't think `CompilerServices.Unsafe` is needed anymore. I believe that was a preview package that we needed during the Core 2.0 era, but it should be available out of the box with .NET Core 3.0+.

**Comment by ameisen** [comment]
So, you were correct. The following two NuGet packages were unneeded or unused:
- System.Runtime.CompilerServices.Unsafe
- SharpFontCore

As per the other NuGet packages:

- Microsoft.NET.Test.Sdk 16.4.0 -> 16.5.0
- NUnit3TestAdapter 3.15.1 -> 3.16.1

I wrote unit tests at my previous job (that was basically my job). I'm a **strong** proponent of keeping the unit test frameworks themselves up to date.

- GtkSharp 3.22.25.56 -> 3.22.25.74
Looks like largely bug fixes. It's a minor revision change so should be no breaking changes.

- FFmpeg.AutoGen 4.2.0 -> 4.2.2.1
Looks like very minor bug fixes and just some spelling changes in the exceptions.
This isn't a very active package, and both Xabe.FFmpeg *and* the _official_ FFmpeg package look like better options.

- DiscordRichPresence 1.0.147 -> 1.0.150
Single version jump, Just added additional logging traces and documentation.
The NuGet packages Discord.Net.* look like they may be better, and are more active.

File: Ryujinx.Common/Ryujinx.Common.csproj [hunk 1]
```diff
@@ -27,10 +27,12 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="JsonPrettyPrinter" Version="1.0.1.1" />
+    <PackageReference Include="JsonPrettyPrinter" Version="1.0.1.1">
+		  <NoWarn>NU1701</NoWarn>
+		</PackageReference>
```
Comment (gdkchan) [inline Ryujinx.Common/Ryujinx.Common.csproj:32]: Indentation is off here.

File: Ryujinx.Common/Ryujinx.Common.csproj [hunk 2]
```diff
@@ -27,10 +27,12 @@
...
     <PackageReference Include="MsgPack.Cli" Version="1.0.1" />
-    <PackageReference Include="Utf8Json" Version="1.3.7" /><PackageReference Include="OpenTK.NetStandard" Version="1.0.5.12" />
-    <PackageReference Include="SharpFontCore" Version="0.1.1" />
+    <PackageReference Include="Utf8Json" Version="1.3.7" />
+    <PackageReference Include="OpenTK.NetStandard" Version="1.0.5.12" />
```
Comment (gdkchan) [inline Ryujinx.Common/Ryujinx.Common.csproj:35]: Why do we need a `OpenTK.NetStandard` reference in `Ryujinx.Common`?

Comment (ameisen) [reply Ryujinx.Common/Ryujinx.Common.csproj]: Good point, removed.

File: Ryujinx.Common/Ryujinx.Common.csproj (hunk 1 above)
Comment (ameisen) [reply Ryujinx.Common/Ryujinx.Common.csproj]: Fixed.

**Comment by jduncanator** [comment]
I'm cautious of the GtkSharp update. I'd really like input from @Thog and @emmauss before touching that package.

**Comment by ameisen** [comment]
> 
> 
> I'm cautious of the GtkSharp update, given how fragile it's been in the past, and the number of custom changes we've made to files from it. I'd really like input from @Thog and @emmauss before touching that package.
> 
> Also, the FFmpeg packages is inconsequential at the moment, it's not used for anything.

If you'd like, I can move the GtkSharp update to its own PR so it can be discussed and analyzed in isolation.

**Comment by ameisen** [comment]
I've moved the GTKSharp update out of the PR for now. I'll resubmit it later in isolation so it can be discussed.

**Pull request merged**